gemelli/scripts/tests/test_standalone_rpca.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import pandas as pd
+from os.path import sep as os_path_sep
+from click.testing import CliRunner
+from skbio import OrdinationResults
+from skbio.util import get_data_path
+from numpy.testing import assert_array_almost_equal
+from gemelli.scripts.__init__ import cli as sdc
+from gemelli.testing import assert_ordinationresults_equal
+
+
+class Test_standalone_rpca(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def test_standalone_rpca_rank_est(self):
+        """Checks the standalone RPCA rank estimate
+           is used instead of a explicit rank
+           setting.
+        """
+        in_ = get_data_path('test.biom', subfolder='rpca_data')
+        out_ = os_path_sep.join(in_.split(os_path_sep)[:-1])
+        runner = CliRunner()
+        result = runner.invoke(sdc.commands['auto-rpca'],
+                               ['--in-biom', in_,
+                                '--output-dir', out_])
+        # Read the results
+        dist_res = pd.read_csv(get_data_path('distance-matrix.tsv',
+                                             subfolder='rpca_data'), sep='\t',
+                               index_col=0)
+        ord_res = OrdinationResults.read(get_data_path('ordination.txt',
+                                                       subfolder='rpca_data'))
+
+        # Read the expected results
+        file_ = 'expected-est-distance-matrix.tsv'
+        dist_exp = pd.read_csv(get_data_path(file_, subfolder='rpca_data'),
+                               sep='\t', index_col=0)
+        ord_exp = OrdinationResults.read(get_data_path(
+                                         'expected-est-ordination.txt',
+                                         subfolder='rpca_data'))
+
+        # Check that the distance matrix matches our expectations
+        assert_array_almost_equal(dist_res.values, dist_exp.values)
+
+        # Check that the ordination results match our expectations -- checking
+        # each value for both features and samples
+        assert_ordinationresults_equal(ord_res, ord_exp)
+
+        # Lastly, check that gemelli's exit code was 0 (indicating success)
+        try:
```

Comment:
This code block is repeated a lot. Possible to encapsulate it?

E.g.,

```
class CliTestCase(unittest.TestCase):

    def assertExitCode(self, value, result):
        try:
            self.assertEqual(0, result.exit_code)
        except:
            ...
```

And then inheriting `CliTestCase` in all of your CLI tests, and calling `assertExitCode`

Reply:
Good idea, this is now implemented. Thanks!